### PR TITLE
Check for placeholders that actually exist in the markup...

### PR DESCRIPTION
**Pull request by ostowe**
...instead of looping through every single module in every template context. Fun fact: for a large style guide which includes examples of all KSS markup (~245 partials loaded as modules!) this feature reduced the number of calls to the `loadModule` function on initial page load from ~75000 to ~270. Page load is visibly faster.

- Also includes improvements to handling of changes to KSS data
- Adds modifier as a JSON data field for use in templates (via `{{ modifier}}`)
- Adds `prototype-[title]` class to default prototype template
- Adds ability to use a custom (not officially supported) `bodyClasses` field for HTMLWebpackPlugin to add custom classes to the prototype body tag
- Alphabetizes methods in the frontend `insertNodes` class

### CODE REVIEW

File: src/js/huron.js [hunk 1]
```diff
@@ -41,15 +41,31 @@ class InsertNodes {
...
    */
   cycleModules(context = false, cached = false, filter = false) {
-    // Reset inserted cache
+    let moduleList = [];
+    let moduleChildren = null;
+
     if (! context) {
       this.inserted = [];
+      moduleList = Object.keys(this._modules);
+      // moduleChildren = [...document.querySelectorAll('[data-huron-id][data-huron-type]')];
```
Comment (pattiereaves) [inline src/js/huron.js]: Keep this?

File: src/js/huron.js [hunk 2]
```diff
@@ -229,9 +254,11 @@ class InsertNodes {
             // Hide the placeholder
             currentTag.style.display = 'none';
 
+            // Set next context
+            hashContext = this.generateHashContext(meta);
+
             // Recursively load modules, excluding the current one
-            // @todo prevent this from calling for each tag?
```
Comment (pattiereaves) [inline src/js/huron.js]: 💃

File: src/js/huron.js [hunk 3]
```diff
@@ -34,22 +33,82 @@ class InsertNodes {
...
+   *
+   * @param {string} modifier - target modifier
+   * @param {object} meta - module metadata
+   */
+  applyModifier(modifier, meta) {
+    let rendered = false;
+    let data = meta.data;
+
+    if (data) {
+      // If we have a modifier, use it, otherwise use the entire data set
```
Comment (pattiereaves) [inline src/js/huron.js]: so we don't need to wrap a single data set in a modifier? sweeet

**Comment by pattiereaves** [comment]
🤖

File: src/js/huron.js (hunk 3 above)
Comment (ostowe) [inline src/js/huron.js]: It's implicit, but yes (based on whether or not you have a `data-huron-modifier` attribute)

**Pull request merged**